// Copyright 2022 ByteDance and/or its affiliates.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";
package idl.matrix.proto;
option java_outer_classname = "LineIdProto";

message MapStringFloatEntry {
  optional string key = 1;
  optional float value = 2;
}
message LineId {
    optional fixed64 uid = 2;
    optional int64 req_time = 3;
    optional fixed64 item_id = 4;
    repeated int32 actions = 6 [packed = true];
    optional int64 chnid = 19;
    repeated int32 pre_actions = 23 [packed = true];
    optional float sample_rate = 27 [default = 1.0];
    repeated int32 special_strategies = 39;
    optional string device_type = 41;
    optional string cid = 48;
    optional string user_id = 49;
    optional bool is_draw = 87;
    optional int32 rank = 145;
    optional string data_source_name = 235;
}
